kyuubi-main/src/test/scala/org/apache/kyuubi/WithKyuubiServerWithMiniYarnService.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi
+
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.internal.Tests.TESTING_HADOOP_CONF_DIR
+import org.apache.kyuubi.server.MiniYarnService
+import org.apache.kyuubi.util.KyuubiHadoopUtils
+
+trait WithKyuubiServerWithMiniYarnService extends WithKyuubiServer {
+  protected val kyuubiServerConf: KyuubiConf
+  protected val connectionConf: Map[String, String]
+  private var miniYarnService: MiniYarnService = _
+
+  override protected lazy final val conf: KyuubiConf = {
+    connectionConf.foreach { case (k, v) =>
+      kyuubiServerConf.set(k, v)
+    }
+    kyuubiServerConf
+  }
+
+  override def beforeAll(): Unit = {
+    miniYarnService = new MiniYarnService()
+    miniYarnService.initialize(new KyuubiConf(false))
+    miniYarnService.start()
+
+    KyuubiHadoopUtils.toSparkPrefixedConf(miniYarnService.getHadoopConf()).foreach { case (k, v) =>
+      conf.set(k, v)
+    }
+    conf.set(TESTING_HADOOP_CONF_DIR, miniYarnService.getHadoopConfDir())
```

Comment:
If this `hadoopDirConf` is empty and we transform configs to kyuubi config, what's the meaning of `HADOOP_CONF_DIR`. Do I missed anything ?

Reply:
just a hardcode check at spark side

Reply:
ah I see.